### ISSUE TITLE
[BUG] Shelly Button1 devices discovered but not added

### ISSUE THREAD

**Issue by essaybee** [StyraHem/ShellyForHASS#521]
## Environment

- ShellyForHASS release with the issue: 0.2.1
- Last working ShellyForHASS release (if known): n/a
- Home Assistant Core release with the issue: 2021.1.5
- Operating environment (Home Assistant/Supervised/Docker/venv):  hass.io

## Describe the bug

I have numerous Shelly devices (Shelly1, Shelly Dimmer, Shelly H&T, Shelly EM) and all work fine. I recently purchased two Shelly Button1 stand alone buttons.  I am using them with USB power and they function normally with Shelly devices but I cannot get them to appear in HA.

## Steps to Reproduce

I added the Shelly Buttons to my wifi network the same as all the others (I do not use Shelly Cloud). 
I tried discover_by_ip but that made no difference.
I have turned on ShellyForHass debug logs and can see that they are discovered by CoAP along with all the other Shelly devices. The are no obvious error messages and I even see multiple "Add Device" messages, but no new devices or entities appear in HA.

## Expected behavior

I expected some new devices and entities to appear in HA

## Traceback/Error logs

In the logs I can also see this kind of entry related to the Shelly Button1 devices:

2021-02-03 14:27:11 DEBUG (CoAP) [pyShelly] http://192.168.1.80/settings
2021-02-03 14:27:11 DEBUG (CoAP) [pyShelly] http://192.168.1.80/settings - Ok

When I retrieved the 'settings' JSON manually and formatted it I was warned that there was a duplicate key for this entry:
  "led_status_disable":false,

Could that be related?

## Additional context

One button is on the factory 1.9.0 firmware (20201029-170348/v1.9.0@85c87178), the other is on the latest upgraded 1.9.0 (20201124-092049/v1.9.0@57ac4ad8)

**Comment by Chris748** [comment]
I had the same problem in the beginning. For me it worked out to configure the button individually and to set a different sensor rule for the specific button device via configuration.yaml

`devices:  #configure devices
    - id: "xxxxxxxxxx"
      sensors:  #Override the global sensor
      - all`

In the end the device shows with a couple of entities/sensors. The only thing I can't get to appear is the switch entity itself. Wich makes the button useless at the moment.

I'm wondering how old is your button? Mine is showing SHBTN-2, shouldn't it be SHBTN-1 instead. So could this be a new hardware version by Allterco?

**Comment by essaybee** [comment]
Hi Chris, thanks for the suggestion. Interestingly that did work for me too - I now have various sensor.shelly_shbtn_2_xxx_xxx entities for the button and a binary_sensor.shelly_shbtn_2_xxxx_mqtt_connected entity too (since I'm using MQTT as a workaround to allow the button to interact with Home Assistant).

So, I'm pretty sure there's still a bug here, as you say, a switch entity is not being auto created for the button which is surely its whole purpose! As I said in my original post (if any of the developers of ShellyForHass are tracking this - could that be due to the duplicate key in the settings?)

My buttons are only a few weeks old and they also show as "shbtn_2".

Fingers crossed that this is an easy fix once someone picks up the defect.

**Comment by essaybee** [comment]
Nope. The MQTT binary_sensor is just a value of "Connected" i.e. just reporting that the switch is connected to the MQTT broker. So, definitely a bug somewhere and there should be another sort of entity that allows actions off the back of the switch

**Comment by hakana** [comment]
I have made some big updates in beta 2 that was released yesterday.

Can you have a try and see if it will work?

**Comment by essaybee** [comment]
Ok, I'm trying that... It took me a while to work out how to install the beta. Your FAQ on where the "Show Beta" flag appears is out of date. I found I have to use the "Reinstall" menu item and then I can choose a ShellyForHASS beta version. I'll post here tonight if it fixes this issue

**Comment by Chris748** [comment]
I gave it a quick try. But sorry to say, on my side no change to the previous behavior.

Tried a couple of things, even deleting and installing Shelly4Hass completely new. Still no auto discovery, no entities showing with only adding default attributes and discovery by ip, after adding device by id and selecting sensors: -all still no switch showing (as before only a couple of sensors like battery and MQTT status).

**Comment by essaybee** [comment]
Yep, I'm seeing the same too, Chris. Still no binary sensors for the buttons. I was going to try and remove the config and go back to pure auto-discovery for the Shelly Button 1's and see if that works but it sounds like you've already tried that.

So, hakan... is there any further debug config we can set to generate further debug logs that would help diagnose the issue, or given the bit I originally posted above, do you think it is possible that the erroneous duplicate keys in the Shelly Button's 'settings' page could be causing this? I still have a bug open with Shelly about that but so far their only (pointless) suggestion is to try removing and re-adding the battery.

**Comment by Chris748** [comment]
Maybe I'm missing something here, but I'm still wondering why my/our entities show as model type SHBTN-2 and for example sensor.shelly_shbtn_2_XXXXXXXX_battery instead of SHBTN-1. 

Shelly is still selling my button version als Button1, but my models are brand new. I even had to wait as they came pretty much directly from the factory, couldn't it be that there is a new, maybe slightly different hardware version? Maybe this is the reason that not everybody seems to have the problems essaybee and myself are fighting with?

**Comment by essaybee** [comment]
hakana, any thoughts on the queries from me and Chris?

**Comment by hakana** [comment]
Does anyone know what the topic for the BTB-2 is in MQTT, shellybutton1 or shellybutton2?

**Comment by essaybee** [comment]
Hi,
I had changed my MQTT prefixes from the default but I just set it back to default and here's a sample of the captured MQTT output:
shellies/shellybutton1-84CCA8ADxxxx/online true
shellies/announce {"id":"shellybutton1-84CCA8ADxxxx","model":"SHBTN-2","mac":"84CCA8ADxxxx","ip":"192.168.1.83","new_fw":false,"fw_ver":"20201124-092049/v1.9.0@57ac4ad8"}
shellies/shellybutton1-84CCA8ADxxxx/announce {"id":"shellybutton1-84CCA8ADxxxx","model":"SHBTN-2","mac":"84CCA8ADxxxx","ip":"192.168.1.83","new_fw":false,"fw_ver":"20201124-092049/v1.9.0@57ac4ad8"}
shellies/shellybutton1-84CCA8ADxxxx/sensor/battery 100
shellies/shellybutton1-84CCA8ADxxxx/sensor/charger true
shellies/shellybutton1-84CCA8ADxxxx/sensor/error 0
shellies/shellybutton1-84CCA8ADxxxx/sensor/act_reasons ["ext_power"]
shellies/shellybutton1-84CCA8ADxxxx/input_event/0 {"event":"SS","event_cnt":8}

**Comment by hakana** [comment]
I will release the new beta hopefully tomorrow

**Comment by hakana** [comment]
Beta 3 now released with support for SHBTN-2. I have not tested this because I do not have this device.

**Comment by essaybee** [comment]
Hi Hakana,

Great news - this works and it automatically detected my two Shelly Button 1's and now added the all important binary_sensor for the switch 👍 

Just in case anyone else is reading this thread... I first deleted the manual config mentioned by Chris above and re-started HA. Interestingly all my old entities (minus the switch sensor) still existing. I then updated HACS to 0.2.2 beta 3 (and restarted again) and then the new entities appeared.

The only minor quirk so far is that the button's battery sensor entities did not automatically appear in my default 'everything' lovelace UI, which is odd, as everything else automatically gets added to that for test purposes. I can see that the battery level entities do exist and are working though so I won't worry about that.
